### PR TITLE
Redo 7z search

### PR DESCRIPTION
**Pull request by ticolensic**

### CODE REVIEW

**Comment by ticolensic** [comment]
I came across a strange bug, which happened the first time I ran it in cmd. After I reopened cmd and ran it, it disappeared. It tried to run "(x86)\7-Zip\7z.exe" which is obviously the lack of quotes. But it downloaded a couple of files already. Moreover, it doesn't return back to the folder it was run from.

So I propose these changes:

 1. add the quotes
 2. fail fast
 3. return to previous folder

File: script/windows/install_packages.bat [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+echo %sevenZipPath%
+
+if not exist "%sevenZipPath%" (
+    if exist "%ProgramFiles%\7-Zip\7z.exe" (
+        set sevenZipPath=%ProgramFiles%\7-Zip\7z.exe
+    )
+)
+
+if not exist "%sevenZipPath%" (
+    echo "Failed to unzip archives because 7-zip is not installed in system. Please unpack all archives in packages internal folders and manually run setup_packages.bat file after"
```
Comment (ihhub) [inline script/windows/install_packages.bat:15]: This message is invalid as we don't download anything before.

File: script/windows/install_packages.bat [hunk 2]
```diff
@@ -1,5 +1,21 @@
 @echo off
 
+set save_cwd=%cd%
+
+for /f "delims=" %%i in ('"where 7z.exe > nul"') do set sevenZipPath=%%i
+echo %sevenZipPath%
```
Comment (ihhub) [inline script/windows/install_packages.bat:6]: Do we need to output this for end user?

File: script/windows/install_packages.bat [hunk 3]
```diff
@@ -1,5 +1,21 @@
 @echo off
 
+set save_cwd=%cd%
+
+for /f "delims=" %%i in ('"where 7z.exe > nul"') do set sevenZipPath=%%i
```
Comment (ihhub) [inline script/windows/install_packages.bat:5]: Please put 7-zip checks after package downloading. Right now we're forcing user to have 7-zip on his machine which is incorrect.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @ticolensic , thanks for changes! Could you please take a look into given comments? We also can't do fast fail if 7-zip is not install in the system as people could use other archivers.

**Comment by ihhub** [comment]
Closing this pull request as the author hasn't replied for many weeks.

**Pull request closed without merging**